docs/interceptortypes.md

```diff
@@ -0,0 +1,56 @@
+<!--
+---
+linkTitle: "InterceptorType"
+weight: 9
+---
+-->
+# InterceptorType
+
+A `InterceptorType` is cluster scoped resource that registers a new Interceptor that 
+can be invoked during the processing of a trigger to modify the behavior or payload of Triggers. The
+custom resource describes how an EventListener can connect to a workload that
+is running the interceptor business logic (and in the future what extra
+paramters the interceptor accepts).
+
+**NOTE**: This doc is a WIP. Please also see the [Interceptors section](./eventlisteners.md#interceptors) in the EventListener doc.
+
+- [Interceptors](#interceptors)
+  - [Syntax](#syntax)
+    - [clientConfig](#clientConfig)
+
+## Syntax
+
+To define a configuration file for an `Interceptor` resource, you can specify
```

Comment:
bikeshed: I kind of like InterceptorConfiguration to be consistent with [ValidatingWebhookConfiguration](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/).

Other good ideas to borrow from admission webhooks we should look into later:
- Version info (e.g. have different routes based on interceptor version)
- Timeout

Reply:
hmmm yeah...this CRD is definitely based on the WebhookConfiguration CRDs....I don't have a strong preference on calling this InterceptorConfguration --> I thought it was a bit confusing in the sense its not clear if it is some interceptor configuration that can be reused within a trigger vs configuration for a new kind of interceptor.

Reply:
I personally don't love the `Configuration` suffix. It makes sense if we have a different `Interceptor` object but a `Configuration` suffix without an `Interceptor` CRD seems weird to me.

No strong opinions though.